### PR TITLE
fix(cardFooter): alignment issue

### LINKED ISSUE #1140: Bug: Issues in the Card component
**Problem**
While rendering card component, if we do not pass `CardFooterLeading` component and try to render `CardFooterTrailing` only, the buttons are rendered on the left rather than right.
I did not render the CardFooterLeading component, hence the buttons were rendered on the left
image (3)

I rendered CardFooterLeading but I did not pass any props, now the buttons do shift to the right
image (4)

Note: This behaviour only effects desktop variant.

**Expected Behaviour**
Ideally the buttons in the footer should be placed on the  bottom right.

### PR DESCRIPTION
**Pull request by sumitd94**
Resolves #1140 

When we do not provide either `footerTrailing` or `footerLeading` component to the `cardFooter` component, then it's always aligned to the right.

More information on this [issue](https://github.com/razorpay/blade/issues/1140)

Added a fix, so that the `footerTrailing` is alway aligned to the right.

> I did not provide the `footerLeading` component

Screenshot 2023-05-04 at 2 44 15 PM

### CODE REVIEW

File: packages/blade/src/components/Card/CardFooter.tsx [hunk 1]
```diff
@@ -40,6 +40,15 @@ const _CardFooter = ({ children, testID }: CardFooterProps): React.ReactElement
     ComponentIds.CardFooterTrailing,
   ]);
 
+  const footerChildrensArray = React.Children.toArray(children);
+
+  const baseBoxJustifyContent =
+    footerChildrensArray.length && footerChildrensArray.length === 2
+      ? 'space-between'
+      : React.isValidElement(footerChildrensArray[0]) && footerChildrensArray[0]?.props?.actions // FooterTrailing component receives actions as props
```
[the remark this answers is not in the document]
Comment (sumitd94) [reply packages/blade/src/components/Card/CardFooter.tsx:48]: That wont work, have given my explanation here - https://github.com/razorpay/blade/issues/1140#issuecomment-1532738114

File: packages/blade/src/components/Card/CardFooter.tsx [hunk 2]
```diff
@@ -40,6 +40,15 @@ const _CardFooter = ({ children, testID }: CardFooterProps): React.ReactElement
     ComponentIds.CardFooterTrailing,
   ]);
 
+  const footerChildrensArray = React.Children.toArray(children);
+
+  const baseBoxJustifyContent =
+    footerChildrensArray.length && footerChildrensArray.length === 2
```
Comment (saurabhdaware) [inline packages/blade/src/components/Card/CardFooter.tsx:46]:
```suggestion
    footerChildrensArray.length === 2
```

this should be enough here right?

Comment (saurabhdaware) [reply packages/blade/src/components/Card/CardFooter.tsx:46]: === 2 itself will ignore any falsy values

File: packages/blade/src/components/Card/CardFooter.tsx [hunk 3]
```diff
@@ -40,6 +40,15 @@ const _CardFooter = ({ children, testID }: CardFooterProps): React.ReactElement
     ComponentIds.CardFooterTrailing,
   ]);
 
+  const footerChildrensArray = React.Children.toArray(children);
+
```
Comment (saurabhdaware) [inline packages/blade/src/components/Card/CardFooter.tsx:44]: Is this an equivalent condition?

```ts
  const footerChildrensArray = React.Children.toArray(children);
  if (!React.isValidElement(footerChildrensArray[0]) {
    throw new Error(`Invalid React Element ${footerChildrensArray}`);
  }

 const baseBoxJustifyContent = footerChildrensArray.length === 2 || !footerChildrensArray[0]?.props?.actions ? 'space-between' : 'end'
```

File: packages/blade/src/components/Card/CardFooter.tsx (hunk 2 above)
Comment (sumitd94) [reply packages/blade/src/components/Card/CardFooter.tsx]: makes sense, thanks

File: packages/blade/src/components/Card/CardFooter.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sumitd94) [reply packages/blade/src/components/Card/CardFooter.tsx]: @anuraghazra Refactored the condition for better readability.
But we still need the check for `props.action`

File: packages/blade/src/components/Card/CardFooter.tsx [hunk 4]
```diff
@@ -40,6 +40,16 @@ const _CardFooter = ({ children, testID }: CardFooterProps): React.ReactElement
     ComponentIds.CardFooterTrailing,
   ]);
 
+  const footerChildrensArray = React.Children.toArray(children);
+  if (!React.isValidElement(footerChildrensArray[0])) {
+    throw new Error(`Invalid React Element ${footerChildrensArray}`);
+  }
+
+  const baseBoxJustifyContent =
+    footerChildrensArray.length === 2 || !footerChildrensArray[0]?.props?.actions
```
Comment (anuraghazra) [inline packages/blade/src/components/Card/CardFooter.tsx:49]: Let's add a comment here explaining why we need this props.actions check.

**Review by anuraghazra — APPROVED** [review]
Update or add tests as necessary.

**Comment by sumitd94** [comment]
> Update or add tests as necessary.

Have already checked, don't think we need any tests update here because it's an alignment issue mostly.

**Review by anuraghazra — APPROVED** [review]

**Review by saurabhdaware — APPROVED** [review]
Looks good to me. Thanks for the contribution 🎉

**Pull request merged**